streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -564,8 +574,16 @@ private void restoreChangelog(final ChangelogMetadata changelogMetadata) {
             return Collections.emptyMap();
 
         try {
-            return restoreConsumer.endOffsets(partitions);
-        } catch (final TimeoutException e) {
+            if (adminClient != null) {
+                final ListOffsetsResult result = adminClient.listOffsets(partitions.stream().collect(
+                        Collectors.toMap(Function.identity(), tp -> OffsetSpec.latest())));
```

Comment:
Should we set `isolation.level` explicitly? -- In case the default if ever changed, we would have a safe guard?

Reply:
Good point, will do.